### ISSUE TITLE
Unit Tests not running due to presumably duplicate symbols in Realm.framework and RealmSwift.framework

### ISSUE THREAD

**Issue by flipchen** [realm/realm-cocoa#2060]
I recently had a lot of issues with my project setup and Cocoapods, so I replaced it with Carthage and also changed my test setup according to what @segiddins suggested in another issue, so that source files are not linked to the test target and are simply imported through the application module.
That's compiling now, but when I try to run the test suite, I get the following errors. Is there something wrong with my project configuration?

```
objc[50724]: Class RLMNotifier is implemented in both /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm and /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/Realm.framework/Realm. One of the two will be used. Which one is undefined.
objc[50724]: Class RLMArray is implemented in both /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm and /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/Realm.framework/Realm. One of the two will be used. Which one is undefined.
objc[50724]: Class RLMSortDescriptor is implemented in both /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm and /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/Realm.framework/Realm. One of the two will be used. Which one is undefined.
objc[50724]: Class RLMCArrayHolder is implemented in both /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm and /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/Realm.framework/Realm. One of the two will be used. Which one is undefined.
objc[50724]: Class RLMArrayLinkView is implemented in both /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm and /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/Realm.framework/Realm. One of the two will be used. Which one is undefined.
objc[50724]: Class RLMMigrationRealm is implemented in both /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm and /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/Realm.framework/Realm. One of the two will be used. Which one is undefined.
objc[50724]: Class RLMMigration is implemented in both /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm and /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/Realm.framework/Realm. One of the two will be used. Which one is undefined.
objc[50724]: Class RLMObject is implemented in both /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm and /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/Realm.framework/Realm. One of the two will be used. Which one is undefined.
objc[50724]: Class RLMDynamicObject is implemented in both /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm and /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/Realm.framework/Realm. One of the two will be used. Which one is undefined.
objc[50724]: Class RLMObjectSchema is implemented in both /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm and /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/Realm.framework/Realm. One of the two will be used. Which one is undefined.
objc[50724]: Class RLMProperty is implemented in both /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm and /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/Realm.framework/Realm. One of the two will be used. Which one is undefined.
objc[50724]: Class RLMNotificationToken is implemented in both /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm and /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/Realm.framework/Realm. One of the two will be used. Which one is undefined.
objc[50724]: Class RLMRealm is implemented in both /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm and /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/Realm.framework/Realm. One of the two will be used. Which one is undefined.
objc[50724]: Class RLMResults is implemented in both /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm and /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/Realm.framework/Realm. One of the two will be used. Which one is undefined.
objc[50724]: Class RLMTableResults is implemented in both /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm and /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/Realm.framework/Realm. One of the two will be used. Which one is undefined.
objc[50724]: Class RLMEmptyResults is implemented in both /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm and /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/Realm.framework/Realm. One of the two will be used. Which one is undefined.
objc[50724]: Class RLMSchema is implemented in both /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm and /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/Realm.framework/Realm. One of the two will be used. Which one is undefined.
objc[50724]: Class RLMListBase is implemented in both /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm and /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/Realm.framework/Realm. One of the two will be used. Which one is undefined.
objc[50724]: Class RLMSwiftSupport is implemented in both /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm and /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/Realm.framework/Realm. One of the two will be used. Which one is undefined.
objc[50724]: Class RLMObjectBase is implemented in both /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm and /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/Realm.framework/Realm. One of the two will be used. Which one is undefined.
objc[50724]: Class RLMObjectUtil is implemented in both /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/RealmSwift.framework/Frameworks/Realm.framework/Realm and /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Bundle/Application/C49C03E8-7881-4DE6-86A8-E786DD019F99/AppName.app/Frameworks/Realm.framework/Realm. One of the two will be used. Which one is undefined.
IDEBundleInjection.c: Error 3587 loading bundle '/Users/flipchen/Development/AppName/DerivedData/AppName/Build/Products/Debug-iphonesimulator/AppNameTests.xctest': The bundle “AppNameTests” couldn’t be loaded because it is damaged or missing necessary resources.
DevToolsBundleInjection environment:
XCInjectDiagnostics: (null)
XCInjectBundleInto: /Users/flipchen/Development/AppName/DerivedData/AppName/Build/Products/Debug-iphonesimulator/AppName.app/AppName
XCInjectBundle: /Users/flipchen/Development/AppName/DerivedData/AppName/Build/Products/Debug-iphonesimulator/AppNameTests.xctest
TestBundleLocation: /Users/flipchen/Development/AppName/DerivedData/AppName/Build/Products/Debug-iphonesimulator/AppNameTests.xctest
TMPDIR: /Users/flipchen/Library/Developer/CoreSimulator/Devices/8FDDF4FF-F808-4AC9-97F4-784CC6487065/data/Containers/Data/Application/36D9CFDA-D553-441F-84D0-5752CE4F665C/tmp
DYLD_LIBRARY_PATH: /Users/flipchen/Development/AppName/DerivedData/AppName/Build/Products/Debug-iphonesimulator
DYLD_INSERT_LIBRARIES: /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/PrivateFrameworks/IDEBundleInjection.framework/IDEBundleInjection
DYLD_FRAMEWORK_PATH: /Users/flipchen/Development/AppName/DerivedData/AppName/Build/Products/Debug-iphonesimulator
DYLD_FALLBACK_LIBRARY_PATH: /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator.sdk/usr/lib
DYLD_FALLBACK_FRAMEWORK_PATH: /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks


*** Test session exited(82) without checking in. Executable cannot be loaded for some other reason, such as a problem with a library it depends on or a code signature/entitlements mismatch. Retry after a Clean build. If you believe this error represents a bug, please attach the log file at /var/folders/bm/hx4t_ds136d117t_pvjf_0h80000gn/T/com.apple.dt.XCTest-status/Session-2015-06-08_11:43:02-gmFbn0.log
```

**Comment by flipchen** [comment]
Any input on this would be greatly appreciated! I'm pretty much stuck with development right now.

**Comment by bmunkholm** [comment]
Maybe some of our Cocoa devs can guess from above. But if you can share the project (maybe privately to help@realm.io), it will likely be easier to help you.

Thanks.

**Comment by segiddins** [comment]
@flipchen it looks to me like you're using an old version of RealmSwift.framework, which attempted to vendor Realm.framework -- we abandoned that approach in 0.93.x. Could you please try re-installing Realm Swift following our instructions at https://realm.io/docs/swift/latest/ with the newest version of Realm? Thanks!

**Comment by flipchen** [comment]
I made sure I'm using the latest version, which seems to be the following:
```
*** Checking out realm-cocoa at "null-string-beta-2"
```
I tried to follow the installation instructions under `Carthage`, but they seem to be out of date. Right now there's no two different frameworks but only one. I added that framework file to `Linked Frameworks and Libraries` and added the run script phase but now I am getting a `no such module 'RealmSwift'` error from the Swift compiler when I try to import with the following import statement:

```
import RealmSwift
```

**Comment by segiddins** [comment]
@flipchen please make sure you're using Carthage v0.7.4 or later, as earlier versions of Carthage had trouble picking up the correct Realm tag

**Comment by flipchen** [comment]
@segiddins I checked, I have 0.7.4 installed.

**Comment by segiddins** [comment]
In that case, what's the contents of your Cartfile? If you're trying to actually use the null string beta, keep in mind that we can't support Carthage installation for it at this time.

**Comment by flipchen** [comment]
My Cartfile looks like this:

```
github "realm/realm-cocoa"
github "Alamofire/Alamofire" >= 1.2

github "Quick/Quick" ~> 0.3.0
github "Quick/Nimble" ~> 0.4.1
```
I think it is according to your installation instructions on `https://realm.io/docs/swift/latest/`

**Comment by segiddins** [comment]
And it's still pulling in the `null-string-beta-2` branch after running `carthage update`? If so, that's really strange, and I have bio idea why that would be happening.

**Comment by flipchen** [comment]
That's correct. It's pulling this specific branch when I run `carthage update` with the Cartfile I showed you. 
I think I'll make use of the offer and send the project to your support under `help@realm.io`.

**Comment by segiddins** [comment]
Could you try changing your cartfile to specify `github "realm/realm-cocoa" >= 0.93.2` ? I think that might fix things

**Comment by jpsim** [comment]
This Carthage issue should be fixed as of 0.7.4.
